src/test/java/org/galatea/starter/utils/http/converter/CsvSerializerTest.java

```diff
@@ -0,0 +1,143 @@
+package org.galatea.starter.utils.http.converter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+
+/**
+ * Many of these tests serve to document certain behaviors of the Jackson CSV serialization library
+ * used in the current implementation of the CsvSerializer class. Many of these tests may need to
+ * change if a different implementation is used.
+ */
+@Slf4j
+public class CsvSerializerTest {
+
+  @Test
+  public void serializeToCsv_basic() throws Exception {
+    BasicClass row1 = new BasicClass("foo", "bar");
+    BasicClass row2 = new BasicClass("baz", null);
+
+    String expected
+        = "field1,field2\n"
+        + "foo,bar\n"
+        + "baz,\n";
+    String actual = CsvSerializer.serializeToCsv(Arrays.asList(row1, row2), BasicClass.class);
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void serializeToCsv_collections() throws Exception {
+    ClassWithCollections row1 = new ClassWithCollections(
+        new String[] {"foo", "bar"}, Arrays.asList("baz", "qux"));
+
+    // Note that the array and list are to-string'ed as item1;item2
+    String expected
+        = "field1,field2\n"
+        + "foo;bar,baz;qux\n";
+    String actual = CsvSerializer.serializeToCsv(
+        Collections.singletonList(row1), ClassWithCollections.class);
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void serializeToCsv_childClass() throws Exception {
+    ChildClass row1 = new ChildClass("foo", "bar", "baz");
+
+    String expected
+        = "field1,field2,field3\n"
+        + "foo,bar,baz\n";
+    String actual = CsvSerializer.serializeToCsv(Collections.singletonList(row1), ChildClass.class);
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void serializeToCsv_childClass2() throws Exception {
+    BasicClass row1 = new ChildClass("foo", "bar", "baz");
+
+    try {
+      CsvSerializer.serializeToCsv(Collections.singletonList(row1), BasicClass.class);
+      fail("A JsonProcessingException was expected but not thrown");
```

Comment:
I consider this a better practice than `@Test(expected = MyException.class)` because this way the test passes only if the exception is thrown by the specific line. A test with `@Test(expected...)` would pass even if the exception was thrown somewhere else in the method, such as a test setup line.